Submitted By: Ken Moffat <ken at linuxfromscratch dot org>
Date: 2017-01-09
Initial Package Version: 6.9.6-3
Upstream Status: N/A
Origin: Self
Description: Allows the libraries of ImageMagick-6 to be installed
without the programs or man-pages, and with versioned pkgconfig files
so that ImageMagick-7 can be installed alongside this.  Do not build
perl here if doing this (the version from ImageMagick-7 will use the
libraries from 7), and edit any cmake programs using this (e.g.
inkscape-0.92) so that they look for the -6 pkgconfig files.

Rediffed for 6.9.9-35 by Chris Clayton, now requires autoreconf.

Internally, ImageMagick-7 is a major change, and the names of the
libraries and the include directories differ.

For packages which use a conventional configure script, pass
IMAGEMAGICK_CFLAGS=-I/usr/include/ImageMagick-6 and
IMAGEMAGICK_LIBS="-lMagickCore-6.Q16HDRI -lMagick++-6.Q16HDRI -lMagickWand-6.Q16HDRI"

In some places I have had to hardcode -6 because I could not persuade
the build system to convert @MAGICK_CORE_VERSION@ in one file or for
the names of the files.

Also, my raw patch deletes wand/Wand.pc.in but gnu patch fails there,
believing it is a reversal (not understood, a git patch would be
easier if we all had git, mentioned because it will probably happen
again if this has to be rediffed for future versions).

diff -Nrup ImageMagick-6.9.9-35/configure.ac ImageMagick-6.9.9-35.fixed/configure.ac
--- ImageMagick-6.9.9-35/configure.ac	2018-02-19 01:12:55.000000000 +0000
+++ ImageMagick-6.9.9-35.fixed/configure.ac	2018-02-19 11:42:14.460241208 +0000
@@ -3964,12 +3964,12 @@ AC_CONFIG_FILES([\
     config/type.xml \
     ImageMagick.spec \
     Magick++/bin/Magick++-config \
-    magick/ImageMagick.pc \
-    Magick++/lib/ImageMagick++.pc \
-    Magick++/lib/Magick++.pc \
+    magick/ImageMagick-6.pc \
+    Magick++/lib/ImageMagick++-6.pc \
+    Magick++/lib/Magick++-6.pc \
     magick/Magick-config \
     magick/MagickCore-config \
-    magick/MagickCore.pc \
+    magick/MagickCore-6.pc \
     magick/version.h \
     Makefile \
     magick.sh \
@@ -3994,9 +3994,9 @@ AC_CONFIG_FILES([\
     utilities/montage.1 \
     utilities/stream.1 \
     wand/MagickWand-config \
-    wand/MagickWand.pc \
+    wand/MagickWand-6.pc \
     wand/Wand-config \
-    wand/Wand.pc ])
+    wand/Wand-6.pc ])
 AC_OUTPUT
 
 # ==============================================================================
diff -Nrup ImageMagick-6.9.9-35/magick/ImageMagick-6.pc.in ImageMagick-6.9.9-35.fixed/magick/ImageMagick-6.pc.in
--- ImageMagick-6.9.9-35/magick/ImageMagick-6.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ ImageMagick-6.9.9-35.fixed/magick/ImageMagick-6.pc.in	2018-02-19 11:42:14.460241208 +0000
@@ -0,0 +1,14 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@/ImageMagick-@MAGICK_MAJOR_VERSION@
+includearchdir=@INCLUDEARCH_DIR@/ImageMagick-@MAGICK_MAJOR_VERSION@
+libname=MagickCore-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@
+ 
+Name: ImageMagick-@MAGICK_MAJOR_VERSION@
+Description: ImageMagick - convert, edit, and compose images (ABI @MAGICK_ABI_SUFFIX@)
+URL: https://www.imagemagick.org
+Version: @PACKAGE_VERSION@
+Cflags:  -I${includearchdir} -I${includedir} @MAGICK_PCFLAGS@
+Libs: -L${libdir} -l${libname}
+Libs.private: -L${libdir} -l${libname} @MAGICK_LIBS@ @MATH_LIBS@
diff -Nrup ImageMagick-6.9.9-35/magick/MagickCore-6.pc.in ImageMagick-6.9.9-35.fixed/magick/MagickCore-6.pc.in
--- ImageMagick-6.9.9-35/magick/MagickCore-6.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ ImageMagick-6.9.9-35.fixed/magick/MagickCore-6.pc.in	2018-02-19 11:42:14.460241208 +0000
@@ -0,0 +1,15 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@/ImageMagick-@MAGICK_MAJOR_VERSION@
+includearchdir=@INCLUDEARCH_DIR@/ImageMagick-@MAGICK_MAJOR_VERSION@
+libname=MagickCore-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@
+moduledir=@libdir@/ImageMagick-@MAGICK_MAJOR_VERSION@.@MAGICK_MINOR_VERSION@.@MAGICK_MICRO_VERSION@/modules-@MAGICK_ABI_SUFFIX@
+
+Name: MagickCore-@MAGICK_MAJOR_VERSION@
+Description: MagickCore - C API for ImageMagick (ABI @MAGICK_ABI_SUFFIX@)
+URL: https://www.imagemagick.org
+Version: @PACKAGE_VERSION@
+Cflags: -I${includearchdir} -I${includedir} @MAGICK_PCFLAGS@
+Libs: -L${libdir} -l${libname}
+Libs.private: -L${libdir} -l${libname} @MAGICK_LIBS@ @MATH_LIBS@
diff -Nrup ImageMagick-6.9.9-35/magick/Makefile.am ImageMagick-6.9.9-35.fixed/magick/Makefile.am
--- ImageMagick-6.9.9-35/magick/Makefile.am	2018-02-19 01:12:55.000000000 +0000
+++ ImageMagick-6.9.9-35.fixed/magick/Makefile.am	2018-02-19 11:57:14.072263401 +0000
@@ -30,8 +30,8 @@ MAGICK_BIN_SCRPTS = \
   magick/MagickCore-config
 
 MAGICK_PKGCONFIG = \
-  magick/ImageMagick.pc \
-  magick/MagickCore.pc \
+  magick/ImageMagick-@MAGICK_MAJOR_VERSION@.pc \
+  magick/MagickCore-@MAGICK_MAJOR_VERSION@.pc \
   magick/ImageMagick-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@.pc \
   magick/MagickCore-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@.pc
 
@@ -467,9 +467,9 @@ MAGICK_EXTRA_DIST = \
 	magick/Magick-config.in \
 	magick/MagickCore-config.in \
 	$(MAGICK_MANS) \
-	magick/ImageMagick.pc.in \
+	magick/ImageMagick-6.pc.in \
 	magick/libMagickCore.map \
-	magick/MagickCore.pc.in \
+	magick/MagickCore-6.pc.in \
 	magick/config.h_vms \
 	magick/mac.c \
 	magick/nt-base.c \
diff -Nrup ImageMagick-6.9.9-35/Magick++/lib/ImageMagick++-6.pc.in ImageMagick-6.9.9-35.fixed/Magick++/lib/ImageMagick++-6.pc.in
--- ImageMagick-6.9.9-35/Magick++/lib/ImageMagick++-6.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ ImageMagick-6.9.9-35.fixed/Magick++/lib/ImageMagick++-6.pc.in	2018-02-19 11:42:14.460241208 +0000
@@ -0,0 +1,14 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@/ImageMagick-@MAGICK_MAJOR_VERSION@
+includearchdir=@INCLUDEARCH_DIR@/ImageMagick-@MAGICK_MAJOR_VERSION@
+libname=Magick++-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@
+ 
+Name: ImageMagick++-@MAGICK_MAJOR_VERSION@
+Description: Magick++ - C++ API for ImageMagick (ABI @MAGICK_ABI_SUFFIX@)
+Version: @PACKAGE_VERSION@
+Requires: MagickWand-@MAGICK_MAJOR_VERSION@
+Libs: -L${libdir} -l${libname}
+Libs.private: -L${libdir} -l${libname} @MAGICK_LIBS@ @MATH_LIBS@
+Cflags: -I${includearchdir} -I${includedir} @MAGICK_PCFLAGS@
diff -Nrup ImageMagick-6.9.9-35/Magick++/lib/Magick++-6.pc.in ImageMagick-6.9.9-35.fixed/Magick++/lib/Magick++-6.pc.in
--- ImageMagick-6.9.9-35/Magick++/lib/Magick++-6.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ ImageMagick-6.9.9-35.fixed/Magick++/lib/Magick++-6.pc.in	2018-02-19 11:42:14.460241208 +0000
@@ -0,0 +1,14 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@/ImageMagick-@MAGICK_MAJOR_VERSION@
+includearchdir=@INCLUDEARCH_DIR@/ImageMagick-@MAGICK_MAJOR_VERSION@
+libname=Magick++-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@
+ 
+Name: Magick++-@MAGICK_MAJOR_VERSION@
+Description: Magick++ - C++ API for ImageMagick (ABI @MAGICK_ABI_SUFFIX@)
+Version: @PACKAGE_VERSION@
+Requires: MagickWand-@MAGICK_MAJOR_VERSION@
+Libs: -L${libdir} -l${libname}
+Libs.private: -L${libdir} -l${libname} @MAGICK_LIBS@ @MATH_LIBS@
+Cflags: -I${includearchdir} -I${includedir} @MAGICK_PCFLAGS@
diff -Nrup ImageMagick-6.9.9-35/Magick++/Makefile.am ImageMagick-6.9.9-35.fixed/Magick++/Makefile.am
--- ImageMagick-6.9.9-35/Magick++/Makefile.am	2018-02-19 01:12:55.000000000 +0000
+++ ImageMagick-6.9.9-35.fixed/Magick++/Makefile.am	2018-02-19 12:01:09.552269210 +0000
@@ -37,8 +37,8 @@ MAGICKPP_MANS_OPT = \
 	Magick++/bin/Magick++-config.1
 
 MAGICKPP_PKGCONFIG_OPT = \
-	Magick++/lib/Magick++.pc \
-	Magick++/lib/ImageMagick++.pc \
+	Magick++/lib/Magick++-@MAGICK_MAJOR_VERSION@.pc \
+	Magick++/lib/ImageMagick++-@MAGICK_MAJOR_VERSION@.pc \
 	Magick++/lib/Magick++-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@.pc \
 	Magick++/lib/ImageMagick++-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@.pc
 
@@ -55,9 +55,9 @@ MAGICKPP_EXTRA_DIST = \
 	Magick++/README \
 	Magick++/bin/Magick++-config.1 \
 	Magick++/bin/Magick++-config.in \
-	Magick++/lib/ImageMagick++.pc.in \
+	Magick++/lib/ImageMagick++-@MAGICK_MAJOR_VERSION@.pc.in \
 	Magick++/lib/libMagick++.map \
-	Magick++/lib/Magick++.pc.in \
+	Magick++/lib/Magick++-@MAGICK_MAJOR_VERSION@.pc.in \
 	Magick++/demo/model.miff \
 	Magick++/demo/smile.miff \
 	Magick++/demo/smile_anim.miff \
diff -Nrup ImageMagick-6.9.9-35/Makefile.am ImageMagick-6.9.9-35.fixed/Makefile.am
--- ImageMagick-6.9.9-35/Makefile.am	2018-02-19 01:12:55.000000000 +0000
+++ ImageMagick-6.9.9-35.fixed/Makefile.am	2018-02-19 12:06:28.608277081 +0000
@@ -155,7 +155,7 @@ pkgconfig_DATA = \
 	$(MAGICKPP_PKGCONFIG)
 
 # create a copy for pc file (ideally the non abi should be symlinked)
-%-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@.pc: %.pc
+%-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@.pc: %-6.pc
 	cp -f $^ $@
 
 # Architecture independent data files installed in the package's data directory
@@ -188,6 +188,8 @@ install-exec-local: $(PERLMAGICK_INSTALL
 
 install-data-local: $(PERLMAGICK_INSTALL_DATA_LOCAL_TARGETS) $(HTML_INSTALL_DATA_TARGETS)
 
+install-libs-only: install-libLTLIBRARIES install-data-am
+
 uninstall-local: $(PERLMAGICK_UNINSTALL_LOCAL_TARGETS) $(HTML_UNINSTALL_DATA_TARGETS)
 
 clean-local: $(PERLMAGICK_CLEAN_LOCAL_TARGETS)
diff -Nrup ImageMagick-6.9.9-35/scripts/Makefile.am ImageMagick-6.9.9-35.fixed/scripts/Makefile.am
--- ImageMagick-6.9.9-35/scripts/Makefile.am	2018-02-19 01:12:55.000000000 +0000
+++ ImageMagick-6.9.9-35.fixed/scripts/Makefile.am	2018-02-19 12:09:25.000281433 +0000
@@ -16,7 +16,7 @@ MAGICKPP_MANS = \
 	Magick++/bin/Magick++-config.1
 
 MAGICKPP_PKGCONFIG = \
-	Magick++/lib/ImageMagick++.pc
+	Magick++/lib/ImageMagick++-6.pc
 
 MAGICKPP_TESTS = \
 	Magick++/tests/exceptions.sh \
@@ -65,7 +65,7 @@ MAGICKPP_EXTRA_DIST = \
 	Magick++/README \
 	Magick++/bin/Magick++-config.1 \
 	Magick++/bin/Magick++-config.in \
-	Magick++/lib/ImageMagick++.pc.in \
+	Magick++/lib/ImageMagick++-6.pc.in \
 	Magick++/demo/model.miff \
 	Magick++/demo/smile.miff \
 	Magick++/demo/smile_anim.miff \
diff -Nrup ImageMagick-6.9.9-35/wand/MagickWand-6.pc.in ImageMagick-6.9.9-35.fixed/wand/MagickWand-6.pc.in
--- ImageMagick-6.9.9-35/wand/MagickWand-6.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ ImageMagick-6.9.9-35.fixed/wand/MagickWand-6.pc.in	2018-02-19 11:42:14.460241208 +0000
@@ -0,0 +1,15 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@/ImageMagick-@MAGICK_MAJOR_VERSION@
+includearchdir=@INCLUDEARCH_DIR@/ImageMagick-@MAGICK_MAJOR_VERSION@
+libname=MagickWand-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@
+ 
+Name: MagickWand-@MAGICK_MAJOR_VERSION@
+Description: MagickWand - C API for ImageMagick (ABI @MAGICK_ABI_SUFFIX@)
+URL: https://www.imagemagick.org
+Version: @PACKAGE_VERSION@
+Requires: MagickCore-@MAGICK_MAJOR_VERSION@
+Cflags: -I${includearchdir} -I${includedir} @MAGICK_PCFLAGS@
+Libs: -L${libdir} -l${libname} 
+Libs.private: -L${libdir} -l${libname} @MAGICK_LIBS@ @MATH_LIBS@
diff -Nrup ImageMagick-6.9.9-35/wand/Makefile.am ImageMagick-6.9.9-35.fixed/wand/Makefile.am
--- ImageMagick-6.9.9-35/wand/Makefile.am	2018-02-19 01:12:55.000000000 +0000
+++ ImageMagick-6.9.9-35.fixed/wand/Makefile.am	2018-02-19 12:13:11.772287027 +0000
@@ -107,8 +107,8 @@ WAND_BIN_SCRPTS = \
 	wand/MagickWand-config
 
 WAND_PKGCONFIG = \
-	wand/MagickWand.pc \
-	wand/Wand.pc \
+	wand/MagickWand-@MAGICK_MAJOR_VERSION@.pc \
+	wand/Wand-@MAGICK_MAJOR_VERSION@.pc \
 	wand/MagickWand-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@.pc \
 	wand/Wand-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@.pc
 
@@ -139,8 +139,8 @@ WAND_EXTRA_DIST = \
 	wand/ChangeLog \
 	wand/libMagickWand.map \
 	wand/MagickWand-config.1 \
-	wand/MagickWand.pc.in \
+	wand/MagickWand-@MAGICK_MAJOR_VERSION@.pc.in \
 	wand/Wand-config.1 \
-	wand/Wand.pc.in
+	wand/Wand-@MAGICK_MAJOR_VERSION@.pc.in
 
 WAND_CLEANFILES =
diff -Nrup ImageMagick-6.9.9-35/wand/Wand-6.pc.in ImageMagick-6.9.9-35.fixed/wand/Wand-6.pc.in
--- ImageMagick-6.9.9-35/wand/Wand-6.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ ImageMagick-6.9.9-35.fixed/wand/Wand-6.pc.in	2018-02-19 12:18:07.216294316 +0000
@@ -0,0 +1,15 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@/ImageMagick-@MAGICK_MAJOR_VERSION@
+includearchdir=@INCLUDEARCH_DIR@/ImageMagick-@MAGICK_MAJOR_VERSION@
+libname=MagickWand-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@
+ 
+Name: MagickWand-@MAGICK_MAJOR_VERSION@
+Description: MagickCore - C API for ImageMagick (ABI @MAGICK_ABI_SUFFIX@)
+URL: https://www.imagemagick.org
+Version: @PACKAGE_VERSION@
+Requires: MagickCore-@MAGICK_MAJOR_VERSION@
+Cflags: -I${includearchdir} -I${includedir} @MAGICK_PCFLAGS@
+Libs: -L${libdir} -l${libname}
+Libs.private: -L${libdir} -l${libname} @MAGICK_LIBS@ @MATH_LIBS@
